docs/content/docs/test/Runner.md

```diff
@@ -17,48 +17,205 @@ It uses an internal TMS (test management system). We are working on support othe
 
-### How to run tests on local emulator
+## How to
+
+Firstly you must apply a plugin TODO Getting started ref
+
+### Run tests on local emulator
 
 1. Run an emulator with 27 API
-1. Run `./gradlew :subprojects:android-test:test-app:instrumentationLocal`
+2. Run `./gradlew :subprojects:android-test:test-app:instrumentationLocal`
+
+### Filter tests for execution
+
+1. {{< expand "Create filter" >}}
+
+```kotlin
+extensions.getByType<GradleInstrumentationPluginConfiguration>().apply {
+    filters.register("filterName") {
+        fromSource.includeByAnnotations(annotations)
+        fromSource.excludeByAnnotations(annotations)
+        fromSource.includeByPrefixes(prefixes)
+        fromSource.excludeByPrefixes(prefixes)
+        
+        // it is internal for Avito. It uses run history from our test-report system.
+        fromRunHistory.excludePreviousStatuses(statuses)
+        fromRunHistory.excludePreviousStatuses(statuses)
+        fromRunHistory.report("reportId") { reportStatuses ->
+            reportStatuses.include(statuses)
+            reportStatuses.exclude(statuses)
+        }
+    }
+}
+```
+
+{{< /expand >}}
+
+2. {{< expand "Add filter to configuration" >}}
+
+```kotlin
+extensions.getByType<GradleInstrumentationPluginConfiguration>().apply {
+    configurationsContainer.register("Local") {
+        filter = "filterName"
+        // else...   
+    }    
+}
+```
+
+{{< /expand >}}
+
+#### Create filter that filters tests by annotations
+
+```kotlin
+extensions.getByType<GradleInstrumentationPluginConfiguration>().apply {
+    filters.register("filterName") {
+        val yourFullyQualifiedAnnotationName = "package.AnnotationClassName"
+        
+        val annotations = setOf(youFullyQualifiedAnnotationName)
+        // will include only tests with at least one annotation
+        fromSource.includeByAnnotations(annotations)
+        // will exclude all tests with at least one annotation
+        fromSource.excludeByAnnotations(annotations)
+    }
+}
+```
+
+#### Create filter that filters tests by prefix or name
+
+```kotlin
+extensions.getByType<GradleInstrumentationPluginConfiguration>().apply {
+    filters.register("filterName") {
+        val packageTestFilter = "testPackage"
+        val classTestFilter = "testPackage.testClass"
+        val fullyQualifiedTestFilter = "testPackage.testClass.testMetod"
+        val prefixes = setOf(packageTestFilter, classTestFilter, fullyQualifiedTestFilter)
+        // will include only tests from package, class and concrete test
+        fromSource.includeByPrefixes(prefixes)
+        
+        // will exclude all tests from package, class and concrete test
+        fromSource.excludeByPrefixes(prefixes)
+    }
+}
+```
+
+#### Create filter that filters tests by statuses from previous run on the same commit
```

Comment:
```suggestion
#### Filter tests by statuses from previous run on the same commit
```

"Create filter ..." is the implementation, "Filtering by ..." is the scenario.
My proposal is to focus on scenario.